### DISCUSSION TITLE
Question(s) about precipitation values

### DISCUSSION THREAD

**Discussion by NickEspo** [weather-gov/api#647] [General]
I am looking for a way to determine from the raw data feed if the forecast precipitation is rain, sleet/hail/frozen rain or snow. This has led to a few questions:

1. Would it be fair to assume that the only way is to see if non-zero values are posted to each of the fields **quantitativePrecipitation**, **iceAccumulation**, and **snowfallAmount**?
2. Am I correct in thinking that the amount in **quantitativePrecipitation** refers only to _rain_, and does not include snow or ice?
3. Lastly, in the case of records with a duration of > 1 hour, does the amount in any of these fields refer to the total amount over that entire duration, or does it reflect the amount predicted for _each hour_ during that duration?

Thanks for any insight!

**Comment by netbymatt** [comment]
I don't know the actual usage of these. But through my observations it appears that:

1. quantitativePrecipation = rain OR (ice + snowfall). Knowing if there is precipitation is just a matter of checking only quantitativePrecipitation. Or alternatively probabilityOfPrecipitation if you'd like to know the likelyhood of precipitation in % instead of an exact value.
2. Incorrect, note the formula in 1. In practice, if there's a value for either rain or snow, I don't show the value for quantitative precipitation in my front end and instead only show ice and snow. Watching these values over the last few winters I've noticed that quantitativePrecipitation always has a value when snow or ice are present.
3. Duration > 1 hour should be read as "Over the next 3 hours it will rain a total of 0.5 inches"

**Comment by sullynole** [comment]
@netbymatt is correct.  quantitativePrecipation is the expected quantity of liquid precipitation accumulated over the period, which includes the liquid equivalent amount for snow and ice.  QPF amounts are usually over 6h... in the case of quantitativePrecipitation the value in API is a total for the specified period.

**Comment by NickEspo** [comment]
Thank you to both @netbymatt and @sullynole.  I _think_ I understand. To determine whether the precip is rain, snow, ice or mixed:

if (**quantitativePrecipitation**  > 0)
    if (**iceAccumulation** > 0) and (**snowfallAmount** > 0)  then it's mixed; otherwise,
    if (**iceAccumulation** > 0) and (**snowfallAmount** = 0) then it's ice;       otherwise,
    if (**iceAccumulation** = 0) and (**snowfallAmount** > 0) then it's snow;   otherwise,
    if (**iceAccumulation** = 0) and (**snowfallAmount** = 0) then it's rain.

It would be more helpful if there was a single "**typeOfPrecipitation**" parameter, but at least it's can still be calculated from the available data.

However, I _really_ wish that _hourly_ **quantitativePrecipitation** values were available (as they are in nearly all other commercial weather sources), since these values can't be accurately derived.  The best we can do -- if the raw data shows "12mm precipitation for hours 12:00-18:00" -- is to divide it evenly and say there's 2mm of precipitation for each of the six hours, which is unlikely to be accurate.

(Barometric pressure is _also_ in most other global weather sources although missing in NWS data, but that would be greedy.)

**Comment by netbymatt** [comment]
Have you looked at the "weather" field at https://api.weather.gov/gridpoints/MLB/26,68

It has descriptive values like this
```
{
    "validTime": "2023-07-25T21:00:00+00:00/PT3H",
    "value": [
        {
            "coverage": "slight_chance",
            "weather": "rain_showers",
            "intensity": "moderate",
            "visibility": {
                "unitCode": "wmoUnit:km",
                "value": null
            },
            "attributes": []
        }
    ]
},
```
